2022q3/bhyve-debug-server-enhancements.adoc

```diff
@@ -0,0 +1,25 @@
+=== bhyve debug server enhancements
+
+Links: +
+link: https://wiki.freebsd.org/SummerOfCode2022Projects/BhyveDebugServerEnhancements [Wiki project page]
+link: https://reviews.freebsd.org/D35826 [Differential]
+Contact: Bojan Novković <bojan.novkovic@kset.org>
+
+The goal of this project was to enhance the functionality of bhyve’s debug server.
+Several existing features related to single-stepping are tied to Intel-specific
+VM mechanisms, which severely impairs bhyve’s debugging functionality on
+other x86 platforms. The first goal dealt with extending single-stepping support to AMD hosts.
+The second goal was to add support for hardware watchpoints using the guest OS's hardware debugging registers.
+
+The project was carried out under Google's Summer of Code program and was finished around the end of
+July. The project's wiki also contains detailed documentation regarding several implemented mechanisms.
+
+The changes can be summarized as follows:
+
+* Support for software breakpoints on AMD SVM,
+* Support for single-stepping AMD SVM guests using the RFLAGS.TF mechanism,
+* Support for placing hardware watchpoints on Intel VMX and AMD SVM guests.
```

Comment:
Audience check: will interested readers understand "SVM", RFLAGS.TF (guessing it's a bit called TF in the 64-bit flags register, but what is it for or at least what does "TF" stand for?), and "VMX?

Reply:
That is a good point, I have toned down the details and hopefully made it more accessible.